### PR TITLE
Add support for row type in Presto queries.

### PR DESCRIPTION
**Pull request by murraystokely**
Summary: Add a mapping to understand the new Row type.

More refactoring work is still required to get the field names, but this at least lets you pull the data without error.

### CODE REVIEW

File: R/extract.data.R [hunk 1]
```diff
@@ -12,6 +12,8 @@ NULL
   data.list <- response.content[['data']]
   if (is.null(data.list)) {
     data.list <- list()
+  } else {
+    print(paste("response.content[['data']] is", data.list))
```
Comment (cosinequanon) [inline R/extract.data.R]: Why do we need to print this? If we actually need to give information to the user then we should use the `message` function

**Comment by cosinequanon** [comment]
@murraystokely thanks for adding this. Do you think you'll be able to work on this more so we can merge it? We definitely need to add some tests. If you are unable to pick this up I can work on it

**Comment by abelborges** [comment]
Hi, guys. Any update on supporting the `row` type?

**Comment by onurfiliz** [comment]
@abelborges , unfortunately properly supporting the row type requires supporting nested data types, which is quite a bit of refactoring. I have some prototype code, but admittedly I did not have time to bring it to completion.

**Comment by abelborges** [comment]
@onurfiliz Thanks for the answer. I understand the pain.

I'm able to fetch query results with nested fields using code like this:
```
db = DBI::dbConnect(RPresto::Presto(), ...)
tb = DBI::dbGetQuery(db, "SELECT nested_field FROM table ...")
```
The nesting is handled with a `list`-typed column in R. What I cannot do is to use dplyr:
```
db = DBI::dbConnect(RPresto::Presto(), ...)
dplyr::tbl(db, "table")
```
which fails with
```
Error in .json.tabular.to.data.frame(data.list, r.types, timezone = timezone) : 
  Unsupported column type for column [1], presto type: "row", R type: "NA"
```
Can anyone tell what is the main difference here?

**Comment by leepface** [comment]
I'm closing this due to inactivity.

**Pull request closed without merging**